packages/expo-constants/src/Constants.server.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  AndroidManifest,
+  AppOwnership,
+  Constants,
+  ExecutionEnvironment,
+  IOSManifest,
+  NativeConstants,
+  PlatformManifest,
+  UserInterfaceIdiom,
+  WebManifest,
+} from './Constants.types';
+import ExponentConstants from './ExponentConstants.web.js';
+
+import type { ExpoConfig } from 'expo/config';
+import type { Manifest as DevLauncherManifest } from 'expo-dev-launcher';
+import type {
+  EmbeddedManifest,
+  EASConfig,
+  ExpoGoConfig,
+  ExpoUpdatesManifest,
+} from 'expo-manifests';
+
+import type { Manifest as UpdatesManifest } from 'expo-updates';
+
+export {
+  AndroidManifest,
+  AppOwnership,
+  Constants,
+  ExecutionEnvironment,
+  IOSManifest,
+  NativeConstants,
+  PlatformManifest,
+  UserInterfaceIdiom,
+  WebManifest,
+};
+
+// Fall back to ExponentConstants.manifest if we don't have one from Updates
```

Comment:
This file can be greatly simplified because ExpoConstants.manifest is something only the client knows about. The server doesn't know what manifest the client is using, unless the client sends that info to the server. I'm not sure the server can otherwise know correct values for things like Constants.expoConfig/expoGoConfig/easConfig/manifest.

Reply:
manifest is embedded by the transformer on web and server platforms.